docs/VirtualMachine.md

```diff
@@ -0,0 +1,160 @@
+# Virtual Machine Design
+
+This document details the workings and design of the virtual machine that executes a Flux script.
+
+## Execution Stack
+
+A Flux script goes through several transformations from text to execution.
+
+1. Flux source code is created via any means.
+2. The parser produces an abstract syntax tree (AST) from the source code.
+3. The semantic analyzer produces a semantic graph from the AST.
+4. The compiler accepts the semantic graph and produces an intermediate representation (IR) to be executed.
+5. The flux virtual machine executes the IR.
+
+NOTE: Should we call it bytecode or IR?
+Should we call it a VM or an interpreter?
```

Comment:
VM if it's an actual virtual machine. To me interpreter means that it's walking the AST and executing along the way.